soda-consumer-scala/src/main/scala/com/socrata/soda2/consumer/impl/LegacyRowDecoder.scala

```diff
@@ -3,35 +3,42 @@ package impl
 
 import java.net.URI
 
-import com.rojoma.json.ast._
-
-import com.socrata.soda2.{UnknownTypeException, ColumnName}
+import com.rojoma.json.ast.{JObject, _}
+import com.rojoma.json.io.{JsonReader, JsonReaderException}
+import com.socrata.soda2.ColumnName
 import com.socrata.soda2.values._
-import org.joda.time.format.{DateTimeFormatterBuilder, ISODateTimeFormat}
-import com.rojoma.json.io.{JsonReaderException, JsonReader}
-import org.joda.time.{DateTimeZone, DateTime}
+import org.joda.time.format.ISODateTimeFormat
+import org.joda.time.{DateTime, DateTimeZone}
 
-private[consumer] class LegacyRowDecoder(datasetBase: URI, rawSchema: Map[ColumnName, String]) extends (JObject => Row) {
-  import LegacyRowDecoder._
+private[consumer] object LegacyRowDecoder {
 
-  val rowDecoder = new RowDecoder(datasetBase, rawSchema.map(conversionForType).toMap)
+  def apply(datasetBase: URI, rawSchema: Map[ColumnName, String]): JObject => Row =
+    sodaify(datasetBase, rawSchema).andThen(new RowDecoder(datasetBase, convertTypes(rawSchema)))
 
-  val schema = rawSchema.map { case (k,v) => k.toString -> v }
+  def k(x: SodaType) = (_: String) => x
```

Comment:
I wouldn't move this away from where it's used.. `k` isn't a very good name (this only exists because otherwise `typeMap`'s construction would be intolerably verbose.